libr/bin/format/elf/elf.c

```diff
@@ -2712,42 +2718,63 @@ static size_t get_next_not_analysed_offset(ELFOBJ *bin, size_t section_vaddr, si
 	return offset;
 }
 
-static size_t populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *relocs, size_t pos, size_t num_relocs) {
-	size_t size, i, j;
-	Elf_(Xword) rel_mode;
+static void populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *relocs, size_t num_relocs, size_t *pos) {
+	size_t p = *pos;
 
 	if (!bin->g_sections) {
-		return pos;
+		return;
 	}
 
-	for (i = 0; !bin->g_sections[i].last; i++) {
-		rel_mode = get_section_mode (bin, i);
-
-		if (!is_reloc_section (rel_mode) || bin->g_sections[i].size > bin->size || bin->g_sections[i].offset > bin->size) {
+	size_t si = 0; // section index
+	for (si = 0; !bin->g_sections[si].last; si++) {
+		RBinElfSection *s = &bin->g_sections[si];
+		Elf_(Xword) rel_mode = get_section_mode (bin, si);
+		if (!is_reloc_section (rel_mode) || s->size > bin->size || s->offset > bin->size) {
 			continue;
 		}
-
+		if (strstr (s->name, "rela")) {
```

Comment:
this is wrong. If the function startswith with `.rela` `get_section_mode` would return `DT_RELA` and `is_reloc_section` would return true already. Was there any case where this was necessary?

Reply:
I think it was parsing it twice.. but i have to recheck